lib/DBIx/Class/Tree/NestedSet.pm

```diff
@@ -148,10 +216,16 @@ sub delete {
             $del_row->($descendant);
         }
 
-        $self->nodes_rs->update({
-            $left  => \"CASE WHEN $left  > $p_rgt THEN $left  - 2 ELSE $left  END",     #"
-            $right => \"CASE WHEN $right > $p_rgt THEN $right - 2 ELSE $right END",     #"
-        });
+        my $diff = $p_rgt - $p_lft + 1;
+
+        $self->nodes_rs->update(
+            {
+                $left  => \"CASE WHEN $left_q  > $p_rgt THEN $left_q  - $diff ELSE $left_q  END",    #"
```

Comment:
ribasushi comment: use `\[ ... ? ..., @vars ]` syntax instead as this allows for statement caching on the DBI side and thus improves performance up the stack.